python/pyspark/sql/connect/dataframe.py

```diff
@@ -478,9 +478,10 @@ def to_jcols(
 
     def hint(self, name: str, *params: Any) -> "DataFrame":
         for param in params:
-            if param is not None and not isinstance(param, (int, str)):
+            if param is not None and not isinstance(param, (int, str, float, list)):
                 raise TypeError(
```

Comment:
Can we add an error test for `test_extended_hint_types` in `python/pyspark/sql/tests/test_dataframe.py` ??